src/shared/machine-id-setup.c

```diff
@@ -148,14 +159,14 @@ int machine_id_setup(const char *root, bool force_transient, sd_id128_t machine_
         }
 
         /* A we got a valid machine ID argument, that's what counts */
-        if (sd_id128_is_null(machine_id)) {
+        if (sd_id128_is_null(machine_id) || machine_id_from_firmware) {
```

Comment:
why use n intermediary bool? why not just do `… || FLAGS_SET(flags, MACHINE_ID_SETUP_FORCE_FIRMWARE)` here?